File: medplum-typescript/packages/sdk/src/model/fhir/RequestGroup_Action.ts
import { Age } from "./Age";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Duration } from "./Duration";
import { Extension } from "./Extension";
import { Period } from "./Period";
import { Range } from "./Range";
import { Reference } from "./Reference";
import { RelatedArtifact } from "./RelatedArtifact";
import { RequestGroup_Condition } from "./RequestGroup_Condition";
import { RequestGroup_RelatedAction } from "./RequestGroup_RelatedAction";
import { Timing } from "./Timing";

export interface RequestGroup_Action {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    prefix: string | null | undefined;
    title: string | null | undefined;
    description: string | null | undefined;
    textEquivalent: string | null | undefined;
    priority: code | null | undefined;
    code: CodeableConcept[] | null | undefined;
    documentation: RelatedArtifact[] | null | undefined;
    condition: RequestGroup_Condition[] | null | undefined;
    relatedAction: RequestGroup_RelatedAction[] | null | undefined;
    timingDateTime: string | null | undefined;
    timingAge: Age | null | undefined;
    timingPeriod: Period | null | undefined;
    timingDuration: Duration | null | undefined;
    timingRange: Range | null | undefined;
    timingTiming: Timing | null | undefined;
    participant: Reference[] | null | undefined;
    type: CodeableConcept | null | undefined;
    groupingBehavior: code | null | undefined;
    selectionBehavior: code | null | undefined;
    requiredBehavior: code | null | undefined;
    precheckBehavior: code | null | undefined;
    cardinalityBehavior: code | null | undefined;
    resource: Reference | null | undefined;
    action: RequestGroup_Action[] | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/RequestGroup_Action.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableRequestGroup_Action.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface RequestGroup_Action {
  Optional<List<RequestGroup_Action>> action();

  Optional<String> textEquivalent();

  Optional<Age> timingAge();

  Optional<List<Extension>> modifierExtension();

  Optional<Range> timingRange();

  Optional<Period> timingPeriod();

  Optional<code> precheckBehavior();

  Optional<code> requiredBehavior();

  Optional<code> groupingBehavior();

  Optional<code> priority();

  Optional<List<Reference>> participant();

  Optional<String> id();

  Optional<String> description();

  Optional<List<RequestGroup_Condition>> condition();

  Optional<Duration> timingDuration();

  Optional<code> cardinalityBehavior();

  Optional<List<CodeableConcept>> code();

  Optional<List<Extension>> extension();

  Optional<code> selectionBehavior();

  Optional<String> timingDateTime();

  Optional<String> title();

  Optional<String> prefix();

  Optional<Timing> timingTiming();

  Optional<List<RequestGroup_RelatedAction>> relatedAction();

  Optional<Reference> resource();

  Optional<CodeableConcept> type();

  Optional<List<RelatedArtifact>> documentation();

  static ImmutableRequestGroup_Action.Builder builder() {
    return ImmutableRequestGroup_Action.builder();
  }
}
